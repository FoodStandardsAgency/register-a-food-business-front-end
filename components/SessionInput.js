const SessionInput = props => (
  <input
    className="sessionDataInput"
    type="text"
    name="sessionData"
    value={props.sessionData}
    readOnly
    hidden
  />
);

export default SessionInput;
