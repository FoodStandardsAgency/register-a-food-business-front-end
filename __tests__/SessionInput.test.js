import SessionInput from "../components/SessionInput";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("<SessionInput />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<SessionInput />);
    expect(wrapper.length).toBe(1);
  });

  it("can receive props", () => {
    const wrapper = mount(
      <SessionInput {...{ sessionData: "test", another: "test2" }} />
    );
    expect(wrapper.props().sessionData).toBe("test");
    expect(wrapper.props().another).toBe("test2");
  });

  it("matches the previous snapshot", () => {
    const tree = renderer.create(<SessionInput />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
