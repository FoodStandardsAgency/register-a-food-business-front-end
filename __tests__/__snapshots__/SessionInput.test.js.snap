// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SessionInput /> matches the previous snapshot 1`] = `
<input
  className="sessionDataInput"
  hidden={true}
  name="sessionData"
  readOnly={true}
  type="text"
  value={undefined}
/>
`;
